### DISCUSSION TITLE
In the latest version, how to get a copy of Tailwindcss full utility styles

### DISCUSSION THREAD

**Discussion by SongYaoLynn** [tailwindlabs/tailwindcss#11240] [Help] [answered]
1. I am currently using the latest version of Tailwindcss, and I want to get the full style of its utilities and generate it into a css file. How should I do it?
2. If compression is required, is the official minify command used?

**Comment by wongjn** [comment]
1. This would not really be possible for v3 – there's an infinite amount of classes possible (due to the arbitrary features) and even without these, if you really wanted every single class possible, you would need to generate every possible combination of the variants, (like `dark:first:hover:focus:md:visited:last:only:…:m-1`) so there would be all the "base" classes plus all these permutations of variants and thus the total number of possible classes would be something like (70![^1] + 1) × number of "base" classes (maths probably not correct here, but something like this). The "real" number would be slightly less since some variants don't make sense together, i.e. `sm:md:lg:xl:m-1` but this obscenely large number should demonstrate the point.
2. Yep!

[^1]: 70 factorial, equal to 11978571669969891796072783721689098736458938142546425857555362864628009582789845319680000000000000000

**Reply by SongYaoLynn** [reply]
For question 1, can v2 version do it?

**Reply by wongjn** [reply, marked as answer]
Yep! V2 has a more limited set of variants and classes in non-JIT mode.
